common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -549,18 +546,59 @@ def test_user_can_revoke_and_no_refund_available(
         assert CourseEnrollment.is_enrolled(self.user, self.course.id)
 
         # Unenroll with Revoke for refund
-        with patch('lms.djangoapps.commerce.signals.handle_refund_entitlement') as mock_refund_handler:
-            REFUND_ENTITLEMENT.connect(mock_refund_handler)
-
-            revoke_url = url + '?is_refund=true'
-            response = self.client.delete(
-                revoke_url,
-                content_type='application/json',
-            )
-            assert response.status_code == 400
-
-            course_entitlement.refresh_from_db()
-            assert not mock_refund_handler.called
-            assert CourseEnrollment.is_enrolled(self.user, self.course.id)
-            assert course_entitlement.enrollment_course_run is not None
-            assert course_entitlement.expired_at is None
+        revoke_url = url + '?is_refund=true'
+        response = self.client.delete(
+            revoke_url,
+            content_type='application/json',
+        )
+        assert response.status_code == 400
+
+        course_entitlement.refresh_from_db()
+        assert CourseEnrollment.is_enrolled(self.user, self.course.id)
+        assert course_entitlement.enrollment_course_run is not None
+        assert course_entitlement.expired_at is None
+
+    @patch('entitlements.api.v1.views.CourseEntitlement.is_entitlement_refundable', return_value=True)
+    @patch('entitlements.api.v1.views.refund_entitlement', return_value=False)
+    @patch("entitlements.api.v1.views.get_course_runs_for_course")
+    def test_user_is_not_unenrolled_on_failed_refund(
+            self,
+            mock_get_course_runs,
+            mock_refund_entitlement,
+            mock_is_refundable
+    ):
+        course_entitlement = CourseEntitlementFactory.create(user=self.user)
+        mock_get_course_runs.return_value = self.return_values
+
+        url = reverse(
+            self.ENTITLEMENTS_ENROLLMENT_NAMESPACE,
+            args=[str(course_entitlement.uuid)]
+        )
+        assert course_entitlement.enrollment_course_run is None
+
+        # Enroll the User
+        data = {
+            'course_run_id': str(self.course.id)
+        }
+        response = self.client.post(
+            url,
+            data=json.dumps(data),
+            content_type='application/json',
+        )
+        course_entitlement.refresh_from_db()
+
+        assert response.status_code == 201
+        assert CourseEnrollment.is_enrolled(self.user, self.course.id)
+
+        # Unenroll with Revoke for refund
+        revoke_url = url + '?is_refund=true'
+        response = self.client.delete(
+            revoke_url,
+            content_type='application/json',
+        )
+        assert response.status_code == 500
+
+        course_entitlement.refresh_from_db()
+        assert CourseEnrollment.is_enrolled(self.user, self.course.id)
+        assert course_entitlement.enrollment_course_run is not None
+        assert course_entitlement.expired_at is None
```

Comment:
Is there a test for what happens when a refund is requested for a non-refundable entitlement?

Reply:
NVM, I see there's one test for that up above.